package/lib/src/ui/components/input_fields/map.dart

```diff
@@ -0,0 +1,111 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_map/flutter_map.dart';
+import 'package:flutter_map/plugin_api.dart';
+import 'package:latlong2/latlong.dart';
+import 'package:geolocator/geolocator.dart';
+
+class MapScreen extends StatefulWidget {
+  final String locationText;
+
+  const MapScreen({Key? key, required this.locationText}) : super(key: key);
+
+  @override
+  State<MapScreen> createState() => _MapScreenState();
+}
+
+class _MapScreenState extends State<MapScreen> {
+  final MapController _mapController = MapController();
+  Position? _selectedPosition;
+  Marker? _userLocationMarker;
+
+  @override
+  void initState() {
+    super.initState();
+
+    if (widget.locationText.isNotEmpty) {
+      final coords = widget.locationText.split(',');
+      final latitude = double.parse(coords[0]);
```

Comment:
Kindly check for the array prior to assigning as if the Coords array has an empty array, then this component will be broken. @Allen7702 and @ericchingalo

Reply:
Thank you, will do